python/ray/data/read_api.py

```diff
@@ -2487,31 +2487,75 @@ def from_arrow_refs(
     )
 
 
+_DATABRICKS_SPARK_DATAFRAM_CHUNK_BYTES = 32 * 1024 * 1024
+
+
 @PublicAPI
 def from_spark(
     df: "pyspark.sql.DataFrame",
     *,
-    parallelism: Optional[int] = None,
+    parallelism: Optional[int] = -1,
```

Comment:
```suggestion
    parallelism: int = -1,
```

Reply:
why do we need to use -1 instead of None?

Reply:
I think we should use -1 as default value, otherwise `_get_num_output_blocks` doesn't work as expected. Other APIs such as `read_parquet` also uses -1 as default value.